molgenis-data-import/src/main/java/org/molgenis/data/export/EmxExportServiceImpl.java

```diff
@@ -0,0 +1,212 @@
+      // Progress per entity plus package sheet + finished message
+      progress.get().setProgressMax(entityTypes.size() + 2);
+    }
+    writePackageSheets(packageSet, writer, progress);
+    writeEntityTypeSheets(entityTypeSet, writer, progress);
+  }
+
+  protected void resolveMetadata(
+      List<EntityType> entityTypes,
+      List<Package> packages,
+      Set<Package> packageSet,
+      Set<EntityType> entityTypeSet) {
+    resolvePackages(packages, packageSet, entityTypeSet);
+    resolveEntityTypes(entityTypes, entityTypeSet);
+  }
+
+  private void writeEntityTypeSheets(
+      Set<EntityType> entityTypes, ExcelWriter writer, Optional<Progress> progressOptional)
+      throws IOException {
+    writeEntityTypes(entityTypes, writer);
+    for (EntityType entityType : entityTypes) {
+      String progressMessage =
+          MessageSourceHolder.getMessageSource()
+              .getMessage(
+                  "emx_export_progress_message",
+                  new Object[] {entityType.getLabel(LocaleContextHolder.getLocale().getLanguage())},
+                  "Downloading: ",
+                  LocaleContextHolder.getLocale());
+      if (progressOptional.isPresent()) {
+        Progress progress = progressOptional.get();
+        progress.status(progressMessage);
+        progress.increment(1);
+      }
+      writeAttributes(entityType.getAllAttributes(), writer);
+      if (!entityType.isAbstract()) {
+        downloadData(entityType, writer);
+      }
+    }
+  }
+
+  private void resolvePackages(
+      List<Package> packages, Set<Package> packageSet, Set<EntityType> entityTypes) {
+    for (Package pack : packages) {
+      resolvePackage(pack, packageSet, entityTypes);
+    }
+  }
+
+  private void resolveEntityTypes(List<EntityType> entityTypes, Set<EntityType> entityTypeSet) {
+    for (EntityType entityType : entityTypes) {
+      checkIfEmxEntityType(entityType);
+      entityTypeSet.add(entityType);
+    }
+  }
+
+  private void resolvePackage(Package pack, Set<Package> packages, Set<EntityType> entityTypes) {
+    packages.add(pack);
+    for (EntityType entityType : pack.getEntityTypes()) {
+      checkIfEmxEntityType(entityType);
+      entityTypes.add(entityType);
+    }
+    for (Package child : pack.getChildren()) {
+      resolvePackage(child, packages, entityTypes);
+    }
+  }
+
+  private void checkIfEmxEntityType(EntityType entityType) {
+    String entityName = entityType.getId();
+    String packageName = entityType.getPackage() != null ? entityType.getPackage().getId() : "";
+    // Entity name should be fully qualified if it resides in a package
+    if (!(packageName.isEmpty() || entityName.startsWith(packageName))) {
+      throw new InvalidEntityIdentifierException(entityName);
+    }
+  }
+
+  private void downloadData(EntityType entityType, ExcelWriter writer) throws IOException {
+    List<Object> headers =
+        StreamSupport.stream(entityType.getAttributes().spliterator(), false)
```

Comment:
Use Guava ```Streams.stream(myStream)``` instead of ```StreamSupport.stream(myStream.spliterator(), false)```
```suggestion
        com.google.common.collect.Streams.stream(entityType.getAttributes())
```